bin/extract-ORFs.py

```diff
@@ -64,19 +79,51 @@ def translations(read):
         def translations(read):
             return read.translations()
 
+    if kozakInfoFile or kozakOnly:
+        if aa:
+            print('Kozak sequences cannot be computed from aa sequences.',
+                  file=sys.stderr)
+            exit()
+        else:
+            from dark.dna import findKozakConsensus
+
+        def writeToKozakOut(kozakread, kozakInfoFile):
+            """
+            Writes out information about a Kozak sequence stored in kozakread
+            to the file name given in kozakInfoFile.
+            """
+            with open(kozakInfoFile, 'a+') as kozakfp:
+                print('Read ID: ' + kozakread.id, file=kozakfp)
+                print('Kozak sequence: ' + kozakread.sequence, file=kozakfp)
+                print('Offset of the Kozak sequence A: ' +
+                      str(kozakread.start + 6), file=kozakfp)
+                print('Quality of the Kozak consensus: ' +
+                      str(kozakread.kozakQuality) + ' %', file=kozakfp)
+
     for read in reads:
         try:
             for translation in translations(read):
-                for orf in translation.ORFs():
-                    # Omit open ORFs if they're unwanted.
-                    if ((orf.openLeft or orf.openRight) and not allowOpenORFs):
-                        continue
-
+                for orf in translation.ORFs(allowOpenORFs):
                     # Check the length requirements, if any.
                     if ((minORFLength is None or len(orf) >= minORFLength) and
-                            (maxORFLength is None or
-                                len(orf) <= maxORFLength)):
-                        write(orf.toString('fasta'))
+                       (maxORFLength is None or len(orf) <= maxORFLength)):
+                        # If Kozak consensus information is wanted:
+                        if kozakInfoFile:
+                            kozakreads = findKozakConsensus(read)
```

Comment:
There's not much point using an extra variable here. About the only time I do this is to break up an otherwise too-long line. I guess it's a matter of taste, but I don't like adding a burden to the reader (e.g., they might wonder if that variable is re-used below, whereas if it doesn't even exist, that's not an issue).  It's also possible to make a hard-to-find bug because later code may one day come to have a `kozakreads` variable that might be initialized in only one half of an `if` statement, and then the unrelated value from up here would still be present.  All tiny things....

Reply:
ti = it
m s = makes sense
mehr = more
ti m s to not use mehr variables than needed... :)